### ISSUE TITLE
unable to deploy the code with azd deploy 

### ISSUE THREAD

**Issue by achu152** [Azure-Samples/azure-search-openai-demo#2077]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [ x] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
> Tried to deploy the code via azd deploy command . however, the output.tar.gz file is not as expected and throwing error 
does azd deploy command takes care of both building the code and deploying it to the apps service ?
### Any log messages given by the failure
>

### Expected/desired behavior
>
Deploying service backend (Running build process)
Deploying service backend (Running build process)
Deploying service backend (Starting runtime process, 1 in progress instances, 0 successful instances)
Deploying service backend (Starting runtime process, 1 in progress instances, 0 successful instances)
Deploying service backend (Starting runtime process, 1 in progress instances, 0 successful instances)
Deploying service backend (Starting runtime process, 1 in progress instances, 0 successful instances)
Deploying service backend (Starting runtime process, 1 in progress instances, 0 successful instances)
Deploying service backend (Starting runtime process, 1 in progress instances, 0 successful instances)
Deploying service backend (Starting runtime process, 1 in progress instances, 0 successful instances)
Deploying service backend (Starting runtime process, 1 in progress instances, 0 successful instances)
Deploying service backend (Starting runtime process, 1 in progress instances, 0 successful instances)
Deploying service backend (Starting runtime process, 1 in progress instances, 0 successful instances)
Deploying service backend (Starting runtime process, 1 in progress instances, 0 successful instances)
Deploying service backend (Starting runtime process, 1 in progress instances, 0 successful instances)
Deploying service backend (Starting runtime process, 1 in progress instances, 0 successful instances)
Deploying service backend (Starting runtime process, 1 in progress instances, 0 successful instances)
Deploying service backend (Fetching endpoints for app service)
  (✓) Done: Deploying service backend
  - Endpoint: https://app-backend-auecacdaps01.azurewebsites.net/

### OS and Version?
> Windows 7, 8 or 10. Linux (which distribution). macOS (Yosemite? El Capitan? Sierra?)
Linux
### azd version?
> run `azd version` and copy paste here.

### Versions
>

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by bnodir** [comment]
> does azd deploy command takes care of both building the code and deploying it to the apps service ?

No, the `azd deploy` command only handles deploying the code to the App Service. To build the code and deploy it, you should use the `azd up` command.

**Comment by svenfeld** [comment]
I might have encountered the same issue. Scaling up the App service helped (backend).

**Comment by achu152** [comment]
@bnodir ; we have restriction to use openai azure instance . and azd up command throws an error because of this policy so we are trying to point to the existing openai service rather than creating a new one via azd up and manually deployed app service as we were not able to continue our deployment via azd up . is there any script that handle only the code build and deploy part.

**Comment by achu152** [comment]
@bnodir the document states 
Deploying again
If you've only changed the backend/frontend code in the app folder, then you don't need to re-provision the Azure resources. You can just run:
if this is the case , shouldnt azd deploy take care of application code build and deploy ?

**Comment by achu152** [comment]
@svenfeld Thank you soo much i can try this out 
am currently using apps service plan  (B1: 1) . can you please let me know what level of scaling done to work this out .

**Comment by svenfeld** [comment]
@achu152 
P0v3 did the job for me.

**Comment by achu152** [comment]
> @achu152 P0v3 did the job for me.

its not working here, tried out

**Comment by bnodir** [comment]
> @bnodir ; we have restriction to use openai azure instance . and azd up command throws an error because of this policy so we are trying to point to the existing openai service rather than creating a new one via azd up and manually deployed app service as we were not able to continue our deployment via azd up . is there any script that handle only the application code backend and frontend app build and deploy part only.

I see your case now. Could you try to check the local build then? 
```
cd .\app
.\start.ps1 
```

**Comment by bnodir** [comment]
> will this work when we have to deploy to a external system?

I haven’t done this myself, so I’m not familiar with all the steps. You might find helpful information in this [GitHub repository](https://github.com/eugenp/tutorials/tree/master/core-java-modules/core-java-networking).

Another option could be to try Azure Container Apps. I haven’t used it myself either, but the steps are detailed in [this document](https://github.com/Azure-Samples/azure-search-openai-demo/blob/0946893fe904cab1e89de2a38c4421e38d508608/docs/azure_container_apps.md). You may need to deploy containers manually due to your policy requirements.

**Comment by pamelafox** [comment]
The ACA deployment is really nice and fast! I hope to default to it in the future, as its so much faster than App Service deployment, but we're waiting for it to have all the same features as App Service (private endpoints, e.g.)
Definitely try out ACA. Great tip @bnodir

**Comment by pamelafox** [comment]
*If you're switching from appservice to ACA, you likely need to delete your App Service/Plan first, so that azd doesnt get confused figuring out which backend to deploy to.

**Comment by pamelafox** [comment]
We moved the default deployment to ACA, and that has improved deployment success generally. Closing this issue. Please open a new issue if you see issues with the ACA deployment.
